tests/test_flows.py

```diff
@@ -2364,6 +2403,44 @@ def my_flow():
 
 
 class TestFlowHooksOnFailure:
+    def test_on_failure_hooks_is_iterable(self):
```

Comment:
I think it might make sense to name the tests like so:
`test_noniterable_hook_raises()`
`test_empty_hook_list_raises()`
`test_noncallable_hook_raises()`